sys/net/application_layer/cord/ep/cord_ep.c

```diff
@@ -218,13 +218,16 @@ static int _discover_internal(const sock_udp_ep_t *remote,
     int res = gcoap_req_init(&pkt, buf, sizeof(buf), COAP_METHOD_GET,
                              "/.well-known/core");
     if (res < 0) {
+        printf("init failed: %d\n", res);
         return CORD_EP_ERR;
     }
     coap_hdr_set_type(pkt.hdr, COAP_TYPE_CON);
     coap_opt_add_uri_query(&pkt, "rt", "core.rd");
     size_t pkt_len = coap_opt_finish(&pkt, COAP_OPT_FINISH_NONE);
-    res = gcoap_req_send(buf, pkt_len, remote, _on_discover, NULL);
+    res = gcoap_req_send_tl(buf, pkt_len, remote, _on_discover, NULL, GCOAP_SOCKET_TYPE_UDP);
```

Comment:
I think you have missed [this line](https://github.com/smartuni/RIOT/blob/24e6ea83c369b5f8ab130b719875d059485b209a/sys/net/application_layer/cord/ep/cord_ep.c#L154)